privacyidea/lib/utils/__init__.py

```diff
@@ -1440,4 +1440,32 @@ def to_list(input):
         return input
     if isinstance(input, set):
         return list(input)
-    return [input]
\ No newline at end of file
+    return [input]
+
+
+def parse_string_to_dict(s, split_char=":"):
+    """
+    This function can parse a string that is formatted like:
+
+       :key1: valueA valueB valueC :key2: valueD valueE
+
+    and return a dict:
+
+       {"key1": ["valueA", "valueB", "valueC"],
+        "key2": ["valueD", "valueE"]
+
```

Comment:
Please mention that there are no whitespaces allowed in the values

Reply:
will do

Reply:
Done in 9afc7703